internal/api/v1/application/upload.go

```diff
@@ -31,6 +48,16 @@ func (hc Controller) Upload(c *gin.Context) apierror.APIErrors {
 	}
 	defer file.Close()
 
+	// TODO: Does this break streaming of the file? We need to get the whole file
+	// before we can check its type
```

Comment:
Since we are only reading 512 bytes from the file and the returning the cursor back to 0, I think this is fine. We are not reading the whole file nor do we store anything on the filesystem.